source/iopipe/json/serialize.d

```diff
@@ -971,6 +1060,125 @@ unittest
     assert(cArr.length == 3);
 }
 
+private void deserializeImplWithPolicy(T, JT, Policy)(
+    ref JT tokenizer,
+    ref T item,
+    ref Policy policy
+) if (is(T == struct) && !isInstanceOf!(JSONValue, T) && !isInstanceOf!(Nullable, T) && !__traits(hasMember, T, "fromJSON"))
+{
+    
+    
+    // Begin deserialization
+    auto visited = policy.onBegin!(JT, T)(tokenizer);
+    
+    // Process fields
+    auto jsonItem = tokenizer.nextSignificant();
+    while(true)
+    {
+        if(jsonItem.token == JSONToken.Comma)
+        {
+            jsonItem = tokenizer.nextSignificant();
+            static if(tokenizer.config.JSON5)
+            {
+                if(jsonItem.token == JSONToken.ObjectEnd)
+                    break;
+            }
+            continue;
+        }
+        
+        // Validate key
+        static if(tokenizer.config.JSON5)
+        {
+            if(jsonItem.token != JSONToken.String && jsonItem.token != JSONToken.Symbol)
+                jsonExpect(jsonItem, JSONToken.String, "Expecting member name of " ~ T.stringof);
+        }
+        else
+            jsonExpect(jsonItem, JSONToken.String, "Expecting member name of " ~ T.stringof);
+        
+        // Get field name
+        auto nameItem = jsonItem;
+        string key = nameItem.data(tokenizer.chain).to!string;
+        
+        // Expect colon
+        jsonItem = tokenizer.nextSignificant()
+            .jsonExpect(JSONToken.Colon, "Expecting colon when parsing " ~ T.stringof);
+        
+        // Let the policy handle this field
+        bool handled = policy.onField!(JT, T)(tokenizer, item, key, visited);
+        
+        if(!handled) {
+            policy.onUnknownField!(JT, T)(tokenizer, key);
+        }
```

Comment:
Just thinking a bit more on this, this looks like basically the policy talking to itself. We can probably move this directly into the `onField` function rather than enforcing the policy to define this (even if it's a no-op).